From bce7870429760c730768f72fafab778f1b4a2067 Mon Sep 17 00:00:00 2001
From: Pavel Grunt <pgrunt@redhat.com>
Date: Tue, 21 Apr 2015 20:16:38 +0200
Subject: [PATCH] virt-viewer-window: Set initial zoom only once

It is not needed to do it more times as it may cause unnecessary resize
events especially when leaving the fullscreen mode

related: rhbz#1206460
(cherry picked from commit 4219a7ec694e45c43315100dceaa5a5a8542cec9)
---
 src/virt-viewer-window.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/virt-viewer-window.c b/src/virt-viewer-window.c
index 7734b5d..96d5757 100644
--- a/src/virt-viewer-window.c
+++ b/src/virt-viewer-window.c
@@ -108,6 +108,7 @@ struct _VirtViewerWindowPrivate {
     gint zoomlevel;
     gboolean fullscreen;
     gchar *subtitle;
+    gboolean initial_zoom_set;
 };
 
 static void
@@ -1272,7 +1273,8 @@ display_show_hint(VirtViewerDisplay *display,
 
     hint = (hint & VIRT_VIEWER_DISPLAY_SHOW_HINT_READY);
 
-    if (hint && virt_viewer_display_get_enabled(display)) {
+    if (!self->priv->initial_zoom_set && hint && virt_viewer_display_get_enabled(display)) {
+        self->priv->initial_zoom_set = TRUE;
         virt_viewer_window_set_zoom_level(self, self->priv->zoomlevel);
     }
 
